### DISCUSSION TITLE
Creating a "embedding" column in the "movies" table to store embeddings.

### DISCUSSION THREAD

**Discussion by sarafarahabadi** [sovelluskontti/project-semantic-search#35] [General]
I have been stuck on this since yesterday and I need some insight. 
I want to add a column to the table to store the embeddings there. Based on my research, it should be FLOAT_VECTOR to be able to do the search properly. and since I am using sentenceTransformers all-MiniLM-L6-v2 model, vector(384) is needed. 

I found this query in this link: 
create table test ( title text, image_vector float_vector knn_type='hnsw' knn_dims='4' hnsw_similarity='l2' );
link: 
https://manticoresearch.com/blog/vector-search/

based on this, I thought the correct syntax would be:
CREATE TABLE movies (
    id BIGINT,
    title TEXT,
    embedding FLOAT_VECTOR KNN_TYPE='hnsw' KNN_DIMS='384' HNSW_SIMILARITY='l2'
);

now I have this error about the KNN library. I can not install it properly. In some resources, it mentions that KNN is for C++ and not Python. but either way, I can not install it properly. 

I would appreciate if you had any insight or resource to help me about this. I keep searching to find a solution myself as well.

**Comment by sarafarahabadi** [comment]
By the way, I have already created the embeddings for the "first_20_movies_noheader.tsv" file and it is saved into another file named" movie_embeddings.csv" 
I am trying to upload this file into the column "embedding" of the database table.

**Comment by nyluntu** [comment]
Hi.

I’m not sure about the correct solution, but could you verify the following:

- Can you verify that columnar storage is installed? I believe vector search requires this to be the table storage engine.
  - I'm not sure which commands you used during installation, but check the project README for installation instructions: https://github.com/manticoresoftware/columnar/.
  - Hint: It would be helpful to include all the installation commands or steps to set up a local development environment for this project.
- I'm not sure if the table needs to be explicitly set to use the columnar storage engine. Maybe not, but I'm uncertain.
- If you encounter an error, it would be helpful to paste the full error message or share a screenshot. This will make it easier for me to find relevant information and understand the context.
- The Manticore Search community has a forum, so check there as well to see if similar issues have been discussed: https://forum.manticoresearch.com/search?q=knn.

Without further searching, I recommend checking these. If you can't find a solution by tomorrow, respond here and explain the steps you have taken to resolve it. If you encounter any issues and manage to resolve them, it would be very helpful to document those as well, including how you solved certain errors, as someone else (or I) may encounter similar issues.

Documenting the issues will help in better understanding them. For example, you could start a discussion and share the solution once the issue is resolved.

**Comment by sarafarahabadi** [comment]
Hi. 
Thanks for the links. I found a possible solution for installing KNN. It has some issues to resolve but I will see how this turns out today.

**Comment by sarafarahabadi** [comment]
Hi, 
Despite all the complications, I could finally upload the embeddings into the database. There is only one problem that the first movie is inserted twice. I will work on it. but the main problem is resolved.

**Comment by sarafarahabadi** [comment]
Hi again. 
the duplicate problem is resolved and I just pushed the new codes to the repo. Next would be to do the search using the embeddings. I will close this discussion since the problem no longer exists.

**Comment by nyluntu** [comment]
Nice to hear! It would be still good to document the issue somehow that what you did to resolve the issue. Could be helpful later.
